icu4c/source/i18n/calendar.cpp

```diff
@@ -494,7 +494,7 @@ class DefaultCalendarFactory : public ICUResourceBundleFactory {
            return nullptr;
         }
 
-        LocaleKey &lkey = (LocaleKey&)key;
+        LocaleKey &lkey = dynamic_cast<LocaleKey&>(const_cast<ICUServiceKey&>(key));
```

Comment:
If I change this line to static_cast<> I got the following runtime error during the test under CFI
```
         TestGenericAPI {
calendar.cpp:497:27: runtime error: control flow integrity check for type 'icu::LocaleKey' failed during base-to-derived cast (vtable address 0x7fc1131fa730)
0x7fc1131fa730: note: invalid vtable
 c1 7f 00 00  a0 7e 0c 13 c1 7f 00 00  80 7b 0c 13 c1 7f 00 00  a0 7b 0c 13 c1 7f 00 00  f0 59 0c 13
              ^ 
calendar.cpp:497:27: note: check failed in ../../lib/libicui18n.so.73, vtable located in ../../lib/libicuuc.so.73
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior calendar.cpp:497:27 in 
ftang@ftang4:~/icu/icu/icu4c/source/test/intltest$ 

```